pkg/controller/node/nodecontroller.go

```diff
@@ -151,8 +157,17 @@ func NewNodeController(
 		metrics.RegisterMetricAndTrackRateLimiterUsage("node_controller", kubeClient.Core().GetRESTClient().GetRateLimiter())
 	}
 
-	if allocateNodeCIDRs && clusterCIDR == nil {
-		glog.Fatal("NodeController: Must specify clusterCIDR if allocateNodeCIDRs == true.")
+	if allocateNodeCIDRs {
+		if clusterCIDR == nil {
+			glog.Fatal("NodeController: Must specify clusterCIDR if allocateNodeCIDRs == true.")
+		}
+		mask := clusterCIDR.Mask
+		// TODO(mqliang): Make pod CIDR mask size configurable.
+		// For now, we assume podCIDR mask size is 24, so make sure the
+		// clusterCIDR mask size is larger than 24.
+		if maskSize, _ := mask.Size(); maskSize > 24 {
```

Comment:
It's because the previous hard code the podCIDR mask size as 24, see https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/node/nodecontroller.go#L303-L313

I'd like make the podCIDR mask size configurable.